iquip/apps/scheduler.py

```diff
@@ -0,0 +1,327 @@
+    """Model for managing the data in the submitted experiment list.
+    
+    Attributes:
+        experimentData: The list of ExperimentView widget.
+    """
+
+    def __init__(self, data: List[ExperimentView], parent: Optional[QWidget] = None):
+        """Overridden."""
+        super().__init__(parent)
+        self.experimentData = data
+
+    def rowCount(self, parent: Optional[QModelIndex] = QModelIndex()) -> int:
+        """Overridden."""
+        return len(self.experimentData)
+
+    def data(self, index: QModelIndex,
+        role: Optional[Qt.ItemDataRole] = Qt.DisplayRole
+    ) -> ExperimentView:
+        """Overridden."""
+        return self.experimentData[index.row()]
+
+    def supportedDropActions(self) -> int:
+        """Overridden."""
+        return Qt.CopyAction | Qt.MoveAction
+
+    def mimeData(self, index: QModelIndex) -> QMimeData:
+        """Fetches the index of the selected element.
+        
+        Args:
+            index: The ModelIndex instance containing 
+              row, column, and etc. values of the selected element.
+        
+        Returns:
+            A QMimeData instance that holds str value of the index.
+        """
+        mime = super().mimeData(index)
+        mime.setText(str(index[0].row()))
+        return mime
+
+    def dropMimeData(self,  # pylint: disable=too-many-arguments
+        mimedata: QMimeData,
+        action: QAction,
+        row: int,
+        column: int,
+        parentIndex: QModelIndex
+    ) -> bool:
+        """Changes the priority of the experiments.
+        
+        Args:
+            mimedata: The QMimeData instance containing str value of the index.
+            action: The QtAction instance classifying the action 
+              (for terminating the function when it is not dropped in the appropriate region)
+            row: The target row that is to be changed with the experiment in mimedata.
+            column: The target column, which is set to zero as it is a QListView.
+            parentIndex: The ModelIndex instance containing
+                         row, column, and etc. values of the target element.
+
+        Returns:
+            True value.
+        """
+        idx = int(mimedata.text())
+        if action == Qt.IgnoreAction:
+            return True
+        taridx = row if row < self.rowCount() else -1
+        row = row if row >= 0 else self.rowCount()
+        if (self.experimentData[idx].arginfo["priority"]
+            != self.experimentData[taridx].arginfo["priority"]):
+            return True
+        if idx > row:
+            self.experimentData = (self.experimentData[:row] + [self.experimentData[idx]] +
+                           self.experimentData[row:idx] + self.experimentData[idx+1:])
+        elif idx < row:
+            self.experimentData = (self.experimentData[:idx] + self.experimentData[idx+1:row] +
+                           [self.experimentData[idx]] + self.experimentData[row:])
```

Comment:
It seems that you want to pick the element at `idx` and insert it at `row`.
How about implement it like this:
```python
if idx != row:
    item = self.experimentData.pop(idx)
    self.experimentData.insert(item, (row - 1) if idx < row else row)
```